src/test/java/com/glencoesoftware/omero/zarr/ZarrPixelBufferTest.java

```diff
@@ -217,6 +219,84 @@ public Path writeTestZarr(
+        Path output = tmpDir.getRoot().toPath().resolve("output.zarr");
+        Path series_path = output.resolve("0");
+        Path img_path = series_path.resolve("0");
+        int[] shape = new int[5];
+        shape[order.indexOf("C")] = sizeC;
+        shape[order.indexOf("T")] = sizeT;
+        shape[order.indexOf("Z")] = sizeZ;
+        shape[order.indexOf("Y")] = sizeY;
+        shape[order.indexOf("X")] = sizeX;
+        ZarrArray array = ZarrArray.create(img_path.toString(), new ArrayParams()
+        .shape(shape)
+        .dataType(com.bc.zarr.DataType.i4)
+        .dimensionSeparator(DimensionSeparator.SLASH)
+        );
+
+        for (int i = 0; i < sizeC; i++) {
+            for (int j = 0; j < sizeT; j++) {
+                for (int k = 0; k < sizeZ; k++) {
+                    int[] plane = getRandomPlane(sizeX, sizeY);
+                    int[] sh = new int[5];
+                    int[] off = new int[5];
+                    sh[order.indexOf("C")] = 1;
+                    sh[order.indexOf("T")] = 1;
+                    sh[order.indexOf("Z")] = 1;
+                    sh[order.indexOf("Y")] = sizeY;
+                    sh[order.indexOf("X")] = sizeX;
+                    off[order.indexOf("C")] = i;
+                    off[order.indexOf("T")] = j;
+                    off[order.indexOf("Z")] = k;
+                    off[order.indexOf("Y")] = 0;
+                    off[order.indexOf("X")] = 0;
+                    array.write(plane, sh, off);
+                }
+            }
+        }
+
+        String rootZattrs = "{\n" + //
+                        "  \"bioformats2raw.layout\" : 3\n" + //
+                        "}";
+        String zgroup = "{\n" + //
+                        "  \"zarr_format\" : 2\n" + //
+                        "}";
+        Files.write(output.resolve(".zattrs"), rootZattrs.getBytes());
+        Files.write(output.resolve(".zgroup"), zgroup.getBytes());
+        Files.write(series_path.resolve(".zgroup"), zgroup.getBytes());
+
+        List<Object> msArray = new ArrayList<>();
+        Map<String, Object> msData = new HashMap<>();
+        Map<String, Object> msMetadata = new HashMap<>();
+        msMetadata.put("method", "loci.common.image.SimpleImageScaler");
+        msMetadata.put("version", "Bio-Formats 6.5.1");
+        msData.put("metadata", msMetadata);
+        msData.put("axes", getAxes(order));
+        msData.put("datasets", getDatasets(1));
+        msData.put("version", "0.1");
+        msArray.add(msData);
+        ZarrGroup z = ZarrGroup.open(output.resolve("0"));
+        Map<String,Object> attrs = new HashMap<String, Object>();
+        attrs.put("multiscales", msArray);
+        z.writeAttributes(attrs);
+        
+        return output;
+    }
+
+    private int[] getRandomPlane(int x, int y) {
+        int[] res = new int[x*y];
+        for (int i = 0; i < res.length; i++) {
+            res[i] = (int) (Math.random() * 255);
```

Comment:
The usage of random bytes makes the troubleshooting harder as it's unclear which plane is being incorrectly read.